provider/datagen/tests/make-testdata.rs

```diff
@@ -55,6 +55,10 @@ fn generate_fs_and_verify_zero_copy() {
 
     let mut options = options::Options::default();
     options.locales = options::LocaleInclude::Explicit(LOCALES.iter().cloned().collect());
+    options.segmenter_models = options::SegmenterModelInclude::Explicit(vec![
+        "thaidict".into(),
+        "Thai_codepoints_exclusive_model4_heavy".into(),
```

Comment:
Issue: It's definitely a breaking change to remove cjdict from icu_testdata. I agree it would be nice to get rid of the JSON file but it's more important that it stays in the postcard file we ship. Let's at least split that to its own PR that we can discuss and not hold up this PR on it.